### ISSUE TITLE
[Bug]:  The launch template created by Terraform defaults to the "EC2-Classic" setting

### ISSUE THREAD

**Issue by MayuriD89** [hashicorp/terraform-provider-aws#30295]
### Terraform Core Version

0.12.31

### AWS Provider Version

3.70.0

### Affected Resource(s)

aws_launch_template

### Expected Behavior

aws_launch_template should add vpc_security_group_ids under the vpc section and not in EC2 classic settings

### Actual Behavior

For aws_launch template in Provider v3.70.0 there are two parameters availability_zone (Placement block) and security_group_names.

If we specify them in terraform, on AWS Console it defaults to EC2-Classic setting which is now deprecated. And on AWS Console it shows an error, but terraform doesn't have any validation for it.

### Relevant Error/Panic Output Snippet

```shell
AWS Console Error - EC2-Classic is no longer supported. Please select the VPC networking platform to continue.
```

### Terraform Configuration Files

```
test-launch-template01 = {
    DataSource = {
      image_id = {
        most_recent = true
        owners      = ["amazon"]
        ami_name    = "amzn2-ami-kernel-5.10-hvm-2.0.20211201.0-x86_64-gp2"
      }
    }
    name                                 = "launch-template01"
    description                          = "Production server for my application"
    update_default_version               = true
    disable_api_termination              = false
    ebs_optimized                        = true
    instance_initiated_shutdown_behavior = "stop"
    instance_type                        = "m3.xlarge"
    kernel_id                            = "aki-f98888"
    ram_disk_id                          = "ari-a009a2a1"
    key_name                             = "key-pair01"
    user_data                            = "./files/http_script.sh"
   security_group_names                 = ["security-group06"]
    block_device_mappings = [
      {
        device_name = "/dev/sdb"
        virtual_name = "ephemeral0"
      },
      {
        device_name  = "/dev/sdd"
        virtual_name = "ephemeral1"
      }
    ]
    capacity_reservation_specification = {
      capacity_reservation_preference = "open"
    }
    cpu_options = {
      core_count       = 1
      threads_per_core = 1
    }
    credit_specification = {
      cpu_credits = "standard"
    }
    elastic_inference_accelerator = {
      type = "eia2.large"
    }
    iam_instance_profile = {
      arn = "arn:aws:iam::8992882992:instance-profile/iam-instance-profile01"
    }
    monitoring = {
      enabled = true
    }
    placement = {
      affinity          = "Host"
      availability_zone = "ap-northeast-1a"
      group_name        = "placement-group01"
      tenancy           = "dedicated"
      partition_number  = 2
    }
    hibernation_options = {
      configured = true
    }
    enclave_options = {
      enabled = false
    }
  }

```

### Steps to Reproduce

terraform apply

### Debug Output

![img_sg](https://user-images.githubusercontent.com/71746353/228288608-2605b067-5a4d-4912-89ea-0896d70e77fc.png)

### Would you like to implement a fix?

None

**Comment by ewbankkit** [comment]
Relates https://github.com/hashicorp/terraform-provider-aws/issues/29480.

**Comment by MayuriD89** [comment]
Ok so I understood that 'security_group_names' parameter is deprecated and no longer supported.

Also we have observed that 'availability_zone' parameter from placement group leads launch template to EC2 classic setting even when we don't specify 'security_group_names'.

If we don't specify 'availability_zone' then it works normally and gets created without any error.

So we think that 'availability_zone' parameter should be **removed** from terraform documentation along with 'security_group_names'.

**Comment by johnsonaj** [comment]
Closed in #30966. Merged to `main` in #31392
